src/network/netdev/xfrm.c

```diff
@@ -0,0 +1,57 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <net/if.h>
+
+#include "netdev/xfrm.h"
+
+static int xfrm_config_verify(NetDev *netdev, const char *filename) {
+        Xfrm *x;
+
+        assert(netdev);
+        assert(filename);
+
+        x = XFRM(netdev);
+
+        if (x->if_id == 0) {
+                log_info("Xfrm NetDev without InterfaceId= in %s. Using ID '0'", filename);
+                x->if_id = 0;
+        }
+
+        return 0;
+}
+
+static int xfrm_fill_message_create(NetDev *netdev, Link *link, sd_netlink_message *message) {
+        int r;
+        int if_idx;
+
+        assert(netdev);
+        assert(message);
+
+        Xfrm *x = XFRM(netdev);
+
+        if (x->independent)
+                if_idx = if_nametoindex("lo"); // can/should we hardcode this to '1'?
+        else
+                if_idx = link->ifindex; // can this be invalid (= 0), validation necessary?
```

Comment:
Maybe, these lines should be conditioned with `link`.
```
assert(link || x->independent);
if (link)
    if_idx = link->ifindex;
else
    if_idx = LOOPBACK_IFINDEX;
```

> // can this be invalid (= 0), validation necessary?

You've already added a validation below :-)

Reply:
Yes but I am not sure if this path needs validation at all. It does not cost much but why do it when its not needed...
I guess both ways are valid so I leave the decision up to you.

Reply:
About the assert. I added it into the conditional block. Is that fine or do you prefer to have asserts in the global function scope?

Reply:
> About the assert. I added it into the conditional block. Is that fine or do you prefer to have asserts in the global function scope?

Current form is OK, I think.